cg/services/file_delivery/fetch_file_service/fetch_delivery_files_service.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import Workflow
+from cg.constants.delivery import PIPELINE_ANALYSIS_TAG_MAP
+from cg.services.file_delivery.fetch_file_service.models import DeliveryFiles
+from cg.store.store import Store
+
+
+class FetchDeliveryFilesService(ABC):
+    """
+    Abstract class that encapsulates the logic required for managing the files to deliver.
+    1. Get a workflow and data delivery option
+    2. Fetch the files based on tags from Housekeeper
+    3. Return the files to deliver
+    """
+
+    @abstractmethod
+    def __init__(self, status_db: Store, hk_api: HousekeeperAPI):
+        self.status_db = status_db
+        self.hk_api = hk_api
```

Comment:
There is no need to specify a constructor for an abstract class

Reply:
Doesn't Python support hybrid classes quite well though? Or is that a big no-no?

Reply:
> There is no need to specify a constructor for an abstract class and mark it as abstract.
> 
 Unless you want to use them in a factory /registry method that assumes a certain format. Then this abstractmethod check if all subclasses have the same constructor is what I understand from it.